cf_units/cf_units.py

```diff
@@ -1871,8 +1914,9 @@ def convert(self, value, other, ctype=FLOAT64):
                         if issubclass(value_copy.dtype.type, np.integer):
                             value_copy = value_copy.astype(
                                 _ctypes2numpy[ctype])
-                        # strict type check of numpy array
-                        if value_copy.dtype.type not in _numpy2ctypes.keys():
+                        # Strict type check of numpy array.
+                        keys = list(_numpy2ctypes.keys())
```

Comment:
You should be able to safely do this, never creating a list of keys: 
```python
if value_copy.dtype.type not in _numpy2ctypes:
    ....
```
same goes for line 1932.